### PR TITLE
circleci: support telemetry

### PR DESCRIPTION
**Pull request by JulesFaucherre**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Why would we enable it if it's off by default?

**Comment by JulesFaucherre** [comment]
This is linked to the pull-request in one of our public repository: https://github.com/CircleCI-Public/circleci-cli/pull/958

**Comment by JulesFaucherre** [comment]
@SMillerDev It isn't off by default, the first time it will run, it will ask you if you approve us collecting telemetry on with CLI or not. In the pr, you can see I added the `CIRCLECI_CLI_TELEMETRY_OPTOUT=1` on the `test` commands to make sure the approval prompt does not appear

File: Formula/circleci.rb [hunk 1]
```diff
@@ -35,13 +36,14 @@ def install
 
   test do
     # assert basic script execution
```
Comment (SMillerDev) [inline Formula/circleci.rb:38]: How about 
```suggestion
    ENV["CIRCLECI_CLI_TELEMETRY_OPTOUT"] = "1"
    # assert basic script execution
```

Comment (JulesFaucherre) [reply Formula/circleci.rb:39]: Sure, thanks for the suggestion. I don't know ruby at all

**Comment by JulesFaucherre** [comment]
To give a bit more context to the PR, the most important part for us is the line
```ruby
      -X github.com/CircleCI-Public/circleci-cli/telemetry.SegmentEndpoint=https://api.segment.io
```
The line will allow the binaries built with it to send telemetry events to our telemetry server `https://api.segment.io`, thus enabling telemetry

**Review by SMillerDev — DISMISSED** [review]

**Pull request closed without merging**